sql/txn.go

```diff
@@ -30,6 +34,15 @@ func (p *planner) BeginTransaction(n *parser.BeginTransaction) (planNode, error)
 // CommitTransaction commits a transaction.
 func (p *planner) CommitTransaction(n *parser.CommitTransaction) (planNode, error) {
 	err := p.txn.Commit()
+	if err != nil {
+		if restartErr, ok := err.(proto.TransactionRestartError); ok {
+			if restartErr.CanRestartTransaction() != proto.TransactionRestart_ABORT {
+				if replyErr := p.txn.Rollback(); replyErr != nil {
+					log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+				}
```

Comment:
Do you think it's worth putting this logic into a method that can be shared with client.Txn.exec()